lib/lambda_wrap/api_gateway_manager.rb

```diff
@@ -151,7 +153,50 @@ def delete_stage(api_id, env)
         puts 'API Gateway stage ' + env + ' does not exist. Nothing to delete.'
       end
     end
-
+
+    # Generate or Update the API Gateway by using the swagger file.
+    # *Arguments*
+    # [api_name] API Gateway name
+    # [local_swagger_file] Path of the local swagger file
+    # [env] Environment
+    def setup_apigateway_by_swagger_file(api_name, local_swagger_file, env)
+
+      #If API gateway with the name is alaredy present then update it else create a new one
+      apis = @client.get_rest_apis(limit: 500).data
+      api = apis.items.select { |a| a.name == api_name }.first
+      swagger_file_content =  File.read(local_swagger_file)
+
+      gateway_response = nil
+      if (api.nil?)

```

Comment:
It's basically the same. You just reuse existing functionality to ensure the api_id exists, and then you only need a PUT instead of the import, so it simplifies your code while achieving the same goal of ensuring that the API is created if it doesn't exist.